packages/gator-permissions-snap/src/utils/string.ts

```diff
@@ -58,3 +58,32 @@ export function shortenAddress(address = '') {
     skipCharacterInEnd: false,
   });
 }
+
+/**
+ * Truncates the decimal part of a numeric string to a specified number of digits.
+ * If the input is not a valid number, returns the original value.
+ *
+ * @param value - The numeric string to truncate (e.g., "123.456789").
+ * @param decimalPlaces - The maximum number of decimal places to keep. Defaults to 8.
+ * @returns The truncated numeric string, or the original value if not a valid number.
+ *
+ * @example
+ * truncateDecimalPlaces("123.456789123", 4); // "123.4567"
+ * truncateDecimalPlaces("100", 4); // "100"
+ * truncateDecimalPlaces("abc", 4); // "abc"
+ */
+export function truncateDecimalPlaces(
+  value: string,
+  decimalPlaces = 8,
+): string {
+  const trimmedValue = value.trim();
+  // Only allow optional minus, digits, optional decimal and digits
+  if (!/^[-+]?\d+(\.\d+)?$/u.test(trimmedValue)) {
+    return value;
+  }
```

Comment:
Is this covering a valid use case? If it's not, maybe we should throw for invalid input?

Reply:
Good point, I’ll update the code to throw for invalid input.